kernel/aster-nix/src/console.rs

```diff
@@ -18,6 +18,7 @@ impl Write for VirtioConsolesPrinter {
     }
 }
 
+/// Prints the formatted arguments to the standard output.
 pub fn _print(args: Arguments) {
     VirtioConsolesPrinter.write_fmt(args).unwrap();
 }
```

Comment:
Should the `virtio-console` be protected with a `SpinLock`, similar to how it has been implemented for the serial port? Or should both the virtio-console and the serial port be protected with the same lock to ensure that the output messages are never interleaved regardless of whether the virtio-console or the serial port is being used?

Reply:
I am unsure about how should we print to the console in `aster-nix`. Theoretically the kernel shouldn't print any and the console is left for the user's standard IO. Locking standard IO is much easier.

I have tried locking the `_print` function, but it seems that the non-reentrant lock here will lead to unwanted behaviors of the pty driver, [failing the tests](https://github.com/asterinas/asterinas/actions/runs/9723926174/job/26839591045). I don't know if it is a serious problem to deal with.

Reply:
> I am unsure about how should we print to the console in aster-nix. Theoretically the kernel shouldn't print any and the console is left for the user's standard IO. Locking standard IO is much easier.

According to the previous design (I'm not sure if it's currently implemented this way), all console IO go through the serial port before initializing `virtio-console`. Afterward, they will go through `virtio-console`. Since `OSTD` does not recognize the presence of `virtio-console`, it can only use the serial port internally. However, `aster-nix` should use `virtio-console` for all console I/O operations, including console I/O from user programs.

> I have tried locking the _print function, but it seems that the non-reentrant lock here will lead to unwanted behaviors of the pty driver, [failing the tests](https://github.com/asterinas/asterinas/actions/runs/9723926174/job/26839591045). I don't know if it is a serious problem to deal with.

This may be a seperate issue. Can you provide the code for the problem? The action seems does not contain the code.